test/io/ably/test/rest/RestAuth.java

```diff
@@ -462,6 +465,230 @@ public Object getTokenRequest(TokenParams params) throws AblyException {
 			fail("auth_authURL_token: Unexpected exception instantiating library");
 		}
 	}
+	
+	/**
+	 * Verify client does not have a clientId data within token details and published messages, 
+	 * when a token initialized with null client id
+	 */
+	@Test
+	public void auth_clientid_null_success() {
+		try {
+			final TestVars testVars = RestSetup.getTestVars();
+
+			// implement callback, using Ably instance with key
+			TokenCallback authCallback = new TokenCallback() {
+				private AblyRest ably = new AblyRest(testVars.createOptions(testVars.keys[0].keyStr));
+				@Override
+				public Object getTokenRequest(TokenParams params) throws AblyException {
+					return ably.auth.requestToken(null, params);
+				}
+			};
+			
+			// create Ably instance without clientId
+			ClientOptions options = testVars.createOptions();
+			options.clientId = null;
+			options.authCallback = authCallback;
+			AblyRest ably = new AblyRest(options);
+			
+			// Fetch token
+			TokenDetails tokenDetails = ably.auth.requestToken(null, null);
+			assertEquals("Auth#clientId is expected to be null", null, tokenDetails.clientId);
+			
+			// Publish message
+			String messageName = "clientless";
+			String messageData = String.valueOf(System.currentTimeMillis());
+			
+			Channel channel = ably.channels.get("test");
+			channel.publish(messageName, messageData);
+			
+			// Fetch published message
+			PaginatedResult<Message> result = channel.history(null);
+			Message[] messages = result.items();
+			Message publishedMessage = null;
+			Message message;
+			
+			for (int i = 0; i < messages.length; i++) {
```

Comment:
Please use the same style as existing code (there are no spaces between `if` and `(`).